rllib/algorithms/dt/tests/test_dt.py

```diff
@@ -0,0 +1,268 @@
+from pathlib import Path
+import os
+import unittest
+from typing import Dict
+
+import gym
+import numpy as np
+
+import ray
+from ray.rllib import SampleBatch
+from ray.rllib.algorithms.dt import DTConfig
+from ray.rllib.offline.json_reader import JsonReader
+from ray.rllib.utils.framework import try_import_tf, try_import_torch
+from ray.rllib.utils.test_utils import check_train_results
+
+tf1, tf, tfv = try_import_tf()
+torch, _ = try_import_torch()
+
+
+def _assert_input_dict_equals(d1: Dict[str, np.ndarray], d2: Dict[str, np.ndarray]):
+    for key in d1.keys():
+        assert key in d2.keys()
+
+    for key in d2.keys():
+        assert key in d1.keys()
+
+    for key in d1.keys():
+        assert isinstance(d1[key], np.ndarray)
+        assert isinstance(d2[key], np.ndarray)
+        assert d1[key].shape == d2[key].shape
+        assert np.allclose(d1[key], d2[key])
+
+
+class TestDT(unittest.TestCase):
+    @classmethod
+    def setUpClass(cls):
+        ray.init()
+
+    @classmethod
+    def tearDownClass(cls):
+        ray.shutdown()
+
+    def test_dt_compilation(self):
+        """Test whether a CRR algorithm can be built with all supported frameworks."""
+
+        rllib_dir = Path(__file__).parent.parent.parent.parent
+        print("rllib dir={}".format(rllib_dir))
+        data_file = os.path.join(rllib_dir, "tests/data/pendulum/large.json")
+        print("data_file={} exists={}".format(data_file, os.path.isfile(data_file)))
+
+        def input_reading_fn(ioctx):
```

Comment:
Can you use the DatasetReader class by default? We essentially don't want to instantiate json reader and gradually get rid of it

Reply:
done